internal/martian/handler.go

```diff
@@ -164,7 +164,7 @@ func (p proxyHandler) handleConnectRequest(ctx *Context, rw http.ResponseWriter,
 		res = p.errorResponse(req, cerr)
 		p.warning(res.Header, cerr)
 	}
-	defer res.Body.Close()
+	defer mustCloseBody(res.Request.Context(), res.Body)
```

Comment:
I believe handler does not need that because the server closes the body close and manages the connections i.e. it's out of our control.

Reply:
Of course!